### DISCUSSION TITLE
Physical measures of labelled entities from patches.

### DISCUSSION THREAD

**Discussion by charleygros** [biigle/core#357] [Q&A]
Hi there,

Context: I'd like to estimate the percentage cover of benthic organisms from the annotations I've made on BIIGLE.

I could use the BIIGLE reports which indicate the radius of each `Circle`. But for some organisms (eg w a not circular shape), it is a over-estimation of the percentage cover.

What I did:
- I've grabbed the Largo patches, using [this process](https://github.com/biigle/community-scripts/tree/master/grab-largo-patches).
- I segmented the organism within the patch, ie background/ foreground segmentation
- I've noticed that all patches are 180 pixels x135 pixels.
- From the annotation reports, I can have the radius of my `Circle` --> Knowing the physical area of the image, I can derive the physical area of the circle.

--> From the patch segmentation, I would like to compute the physical area of the organism, which is <= `circle_area`.

My question is: what is the relation between the patch size and the `Circle` size? eg width of the patch = radius of the `Circle`.

From this I could:
- organism covers X % of the patch
- so it represents X'% of the `Circle`
- so it covers `circle_area * (1 - X' / 100)` mm2 

Many thanks in advance

**Comment by charleygros** [comment]
Hi @mzur, Would you be able to help me with this? Cheers

**Reply by mzur** [reply]
@charleygros Did this answer your question?

**Reply by charleygros** [reply]
Thank you very much for your insights @mzur !
I'll give it a go and come back to you shortly, Cheers

**Comment by mzur** [comment]
Sorry, I wasn't notified of this issue because GitHub changed my notification settings.

It might be a little complicated to go from the annotation patch position and resolution back to the original position and resolution. It gets easier if you can ignore the position and only want the resolution, which seems to be the case.

The patches are generated [here](https://github.com/biigle/largo/blob/1be2a7524e3a2e893142af5721521bc9b086c7cd/src/Jobs/GenerateAnnotationPatch.php#L255-L279). To get the original resolution, you have to reverse the [`resize`](https://github.com/biigle/largo/blob/1be2a7524e3a2e893142af5721521bc9b086c7cd/src/Jobs/GenerateAnnotationPatch.php#L272) operation, i.e. you have to determine `$thumbWidth` and  `$rect['width']`. `$thumbWidth` is configured [here](https://github.com/biigle/core/blob/51398dabfa4cffeaa46e0d764ad603992b755840/config/thumbnails.php#L15). `$rect['width']` is calculated [here](https://github.com/biigle/largo/blob/1be2a7524e3a2e893142af5721521bc9b086c7cd/src/Jobs/GenerateAnnotationPatch.php#L215) and may be modified [here](https://github.com/biigle/largo/blob/1be2a7524e3a2e893142af5721521bc9b086c7cd/src/Jobs/GenerateAnnotationPatch.php#L240).

**Comment by mzur** [comment]
It may actually be easier to take the circle annotations and original images and then extract new image patches in the original resolution for your segmentation.
